### PR TITLE
Buttons: fixed up Scary styles

### PR DESCRIPTION
**Pull request by rickybanister**
* I started out just trying to fix the hover state for a disabled scary button—it was showing a red border on hover despite being disabled.
* I then noticed a bunch of deprecated styles in the buttons css so I removed them. They affected site-settings delete site stuff, so I also updated that page to use the `Button` component properly, removed noticons, and used the new `scary` prop instead of the old `is-destructive` and `is-dangerous` classes.
* I also made the border color darker for primary scary buttons when hovering
* I also added a hover style for borderless scary buttons

**Before:**

screen shot 2015-12-08 at 10 42 46 am
screen shot 2015-12-08 at 10 43 27 am

**After:**

screen shot 2015-12-08 at 10 39 52 am
screen shot 2015-12-08 at 10 39 59 am
screen shot 2015-12-08 at 10 40 40 am
screen shot 2015-12-08 at 10 41 34 am
screen shot 2015-12-08 at 10 41 40 am

cc @adambbecker since you did the original work on settings
cc @mtias 

- [ ] Add external icon back to the Export Button, but conditionalize it so it disappears when export launches

### CODE REVIEW

**Comment by rickybanister** [comment]
I went through the other buttons in delete site and they should all now use gridicons:

Before:
screen shot 2015-12-08 at 10 50 51 am

After:
screen shot 2015-12-08 at 11 02 38 am

**Comment by rickybanister** [comment]
I also moved the actual remaining deprecated styles to the bottom of the stylesheet, moving the `is-borderless` stuff above that for clarity.

File: client/my-sites/site-settings/start-over/index.jsx [hunk 1]
```diff
@@ -57,7 +59,13 @@ module.exports = React.createClass( {
 						}</p>
 					</ActionPanelBody>
 					<ActionPanelFooter>
-						<a className="button" href="https://en.support.wordpress.com/contact" target="_blank">{ strings.contactSupport }<span className="noticon noticon-external settings-action-panel__footer-button-icon"></span></a>
+						<Button
+							className="settings-action-panel__support-button"
```
Comment (mtias) [inline client/my-sites/site-settings/start-over/index.jsx]: this should be `start-over__support-button`

**Comment by mtias** [comment]
This looks good to me.

File: client/my-sites/site-settings/delete-site/index.jsx [hunk 2]
```diff
@@ -94,15 +96,15 @@ module.exports = React.createClass( {
 						}</p>
 					</ActionPanelBody>
 					<ActionPanelFooter>
-						<a
-							className="button"
+						<Button
+							className="settings-action-panel__export-button"
```
Comment (mtias) [inline client/my-sites/site-settings/delete-site/index.jsx]: these classes should be fixed. Not sure if you want to do so now.

Comment (mtias) [inline client/my-sites/site-settings/delete-site/index.jsx]: In this case the component folder needs to be renamed.

**Comment by rickybanister** [comment]
I'm going to refactor the class names and folder structure in a separate pr.

**Comment by MichaelArestad** [comment]
Super late LGTM.

**Pull request merged**